scripts/build.py

```diff
@@ -36,7 +36,7 @@
 from conda_build.metadata import MetaData
 from conda_build.variants import get_package_variants
 from conda_build.render import distribute_variants
-from conda_build.build import build as CondaBuild
+from conda_build.api import build as CondaBuild
```

Comment:
I just took another look at `conda_build.api.build` and I am not sure if it's better to call it directly. Internally, it calls `conda_build.build.build_tree`, which does a whole lot of work (i.e., resolving dependencies) that we already did before calling `CondaBuild` (i.e., `conda_build.api.build`). In addition,  it tests not only the built package itself, but also all "downstream" packages --- not sure what it means but I don't think we need this behavior, do we? See https://github.com/conda/conda-build/blob/fc22ab8c161a6c79ed3d48fab0ffc954a5fa61c2/conda_build/build.py#L2282-L2323

Have you tested this change with the MPI chain and Matplotlib? Are they working fine? If so, we can merge it and worry about the redundant work later when the build system is reworked in the summer. 

Alternatively, I think we could just call `conda_build.build.test` ourselves after the `CondaBuild` block https://github.com/NSLS-II/lightsource2-recipes/blob/ad8b4b88b6640134ab0e1923a3af349ff86e6321/scripts/build.py#L537-L553
```python
        try:
            conda_build.build.test(meta, config=meta.config.copy(), stats=stats)
        except:
            # report error, skip this package, and proceed to the next one?
```
which is similar to how it is used in `conda_build.build.build_tree`, see https://github.com/conda/conda-build/blob/fc22ab8c161a6c79ed3d48fab0ffc954a5fa61c2/conda_build/build.py#L2277-L2281

Reply:
Thanks @leofang. Yes, I tried to add the step for testing explicitly, but then decided to follow what conda-build does. I haven't tried it with mpi or matplotlib yet. Will test and let you know.

Reply:
right, they are the two most weird recipes we have if I'm not mistaken, so should be good enough test cases 😄